docs/installation/advanced/integrating-from-source-vite.md

```diff
@@ -0,0 +1,266 @@
+```
+ 
+ ## Installing necessary dependencies
+
+After initializing the project, you can start installing packages. Fundamentally, you need three things: an editor base, editor plugins, and a Vite plugin for CKEditor. This example will use Classic Editor. In terms of plugins, you can use whatever you want. If you need inspiration, you can base your integration on one of the existing builds. There is a list of packages in the [classic build's](https://github.com/ckeditor/ckeditor5/blob/master/packages/ckeditor5-build-classic/package.json) `package.json` file. But remember that all packages (excluding `@ckeditor/ckeditor5-dev-*`) must have the same version as the base editor package.
+
+You can install packages using npm like `npm install @ckeditor/ckeditor5-editor-classic`, or copy dependencies from the build repo and typing `npm install`. The `dependencies` section of your `package.json` should look similar to this:
+
+```js
+"dependencies": {
+	// Dependencies.
+	// ...
+
+    "@ckeditor/ckeditor5-adapter-ckfinder": "^x.y.z",
+    "@ckeditor/ckeditor5-autoformat": "^x.y.z",
+    "@ckeditor/ckeditor5-basic-styles": "^x.y.z",
+    "@ckeditor/ckeditor5-block-quote": "^x.y.z",
+    "@ckeditor/ckeditor5-easy-image": "^x.y.z",
+    "@ckeditor/ckeditor5-editor-classic": "^x.y.z",
+    "@ckeditor/ckeditor5-essentials": "^x.y.z",
+    "@ckeditor/ckeditor5-heading": "^x.y.z",
+    "@ckeditor/ckeditor5-image": "^x.y.z",
+    "@ckeditor/ckeditor5-link": "^x.y.z",
+    "@ckeditor/ckeditor5-list": "^x.y.z",
+    "@ckeditor/ckeditor5-paragraph": "^x.y.z",
+    "@ckeditor/ckeditor5-theme-lark": "^x.y.z",
+    "@ckeditor/ckeditor5-upload": "^x.y.z"
+
+    // More dependencies.
+	// ...
+}
+```
+
+## Vite configuration
+
+When your editor has all the necessary plugins, you can move on to integration with Vite. There is also an official plugin for this purpose. It handles loading the SVG icons and styles from the packages and the theme package. You can install it via the below command.
+
+```js
+npm install --save @ckeditor/ckeditor5-vite-plugin
+```
+
+The plugin is installed but will not work yet, so you need to add it to the Vite configuration. First, add the `vite.config.js` file at the root of your project (or use an existing one). Then, modify the file by adding the following lines of code.
+
+```js
+// vite.config.js
+
+import { defineConfig } from 'vite';
+import ckeditor5 from '@ckeditor/vite-plugin-ckeditor5';
+
+export default defineConfig( {
+	plugins: [
+		ckeditor5( { theme: require.resolve( '@ckeditor/ckeditor5-theme-lark' ) } )
+	]
+} );
+```
+
+The configuration slightly differs for ESM projects. If you try to start the dev server using the `npm run dev` command, you may encounter an error: `require.resolve is not a function`. In this case, you need some additional lines of code.
+
+```js
+// vite.config.js
+
+import { createRequire } from 'node:module';
+const require = createRequire( import.meta.url );
+```
```

Comment:
This section about workaround in ESM projects is mentioned only in this guide but the issue can occur in React, Vue2 and Vue 3 integrations. Do you think we should mention that in other guides as well?